src/app/icd/client/CheckInDelegate.h

```diff
@@ -36,9 +39,40 @@ class DLL_EXPORT CheckInDelegate
      * @brief Callback used to let the application know that a check-in message was received and validated.
      *
      * @param[in] clientInfo - ICDClientInfo object representing the state associated with the
-                               node that sent the check-in message.
+     *                         node that sent the check-in message.
      */
     virtual void OnCheckInComplete(const ICDClientInfo & clientInfo) = 0;
+
+    /**
+     * @brief Callback used to let the application know that a key refresh is
+     * needed to avoid counter rollover problems.
+     *
+     * The implementer of this function should create a new RefreshKeySender object. They should also generate a new key and store
+     * it in the RefreshKeySender object. This object will be tied to the specific key refresh process and will not be used by the
+     * caller after that particular key refresh process has ended, regardless of success or failure.
```

Comment:
Nothing either here or on RefreshKeySender makes it clear how the "store it" step should work.

Why is the new key not just a constructor argument to RefreshKeySender?

Reply:
Modified the code to pass the new key as an argument to the constructor.

Reply:
Modified the code and made the new key an argument to RefreshKeySender.